network/p2p/connection/connector.go

```diff
@@ -132,28 +131,24 @@ func (l *Libp2pConnector) pruneAllConnectionsExcept(peerIDs peer.IDSlice) {
 		}
 
 		peerInfo := l.host.Network().Peerstore().PeerInfo(peerID)
-		log := l.log.With().Str("remote_peer", peerInfo.String()).Logger()
-		if l.host.ConnManager().IsProtected(peerID, "") {
-			log.Trace().Msg("skipping pruning since connection is protected")
-			continue // connection is protected (stream or connection in progress), skip pruning
-		}
+		lg := l.log.With().Str("remote_peer", peerInfo.String()).Logger()
+
+		// log the protected status of the connection
+		protected := l.host.ConnManager().IsProtected(peerID, "")
+		lg = lg.With().Bool("protected", protected).Logger()
 
-		// retain the connection if there is a Flow One-to-One stream on that connection
-		// (we do not want to sever a connection with on going direct one-to-one traffic)
+		// log if any stream is open on this connection.
 		flowStream := p2putils.FlowStream(conn)
 		if flowStream != nil {
-			log.Info().
-				Str("stream_protocol", string(flowStream.Protocol())).
-				Msg("skipping connection pruning with peer due to one-to-one stream")
-			continue // flow stream found, skip pruning
+			lg = lg.With().Str("flow_stream", string(flowStream.Protocol())).Logger()
```

Comment:
We no longer should enforce the policy of _not pruning the protected connection_. We only trust the `peerIDs` (provided by the peer manager), and _purne_ all connections except to those coming from `peerIDs`.  We only log the protected status of a connection (indicating whether it was protected by libp2p side).

Reply:
It seems like libp2p may not be re-using streams. If that's the case then it could be that a stream only exists when a node is actively sending or receiving a direct message. Is there a way to check if the stream is actively being used? Why did we remove skipping pruning connections if stream is not nil.